build/source/netcdf/read_icond.f90

```diff
@@ -185,71 +186,79 @@ subroutine read_icond(iconFile,                      & ! name of initial conditi
  err = nf90_inq_dimid(ncID,"hru",dimID);               if(err/=nf90_noerr)then; message=trim(message)//'problem finding hru dimension/'//trim(nf90_strerror(err)); return; end if
  err = nf90_inquire_dimension(ncID,dimID,len=fileHRU); if(err/=nf90_noerr)then; message=trim(message)//'problem reading hru dimension/'//trim(nf90_strerror(err)); return; end if
 
- ! is there any snow in initial conditions file?
- snowExists = .true.
- err = nf90_inq_dimid(ncID,trim(midSnowDimName),dimID)
- if (err/=0) snowExists = .false.
-
  ! loop through prognostic variables
  do iVar = 1,size(prog_meta)
 
-  ! if there is no snow, do not look for snow variables
-  if ((.not.snowExists).and.((prog_meta(iVar)%varType==iLookVarType%midSnow).or.(prog_meta(iVar)%varType==iLookVarType%ifcSnow))) then
-   do iGRU = 1,nGRU
-    do iHRU = 1,gru_struc(iGRU)%hruCount
-     progData%gru(iGRU)%hru(iHRU)%var(iVar)%dat = 0.
-    end do
-   end do
-   cycle
-  end if
+  ! skip variables that are computed later
+  if(prog_meta(iVar)%varName=='scalarCanopyWat'           .or. &
+     prog_meta(iVar)%varName=='spectralSnowAlbedoDiffuse' .or. &
+     prog_meta(iVar)%varName=='scalarSurfaceTemp'         .or. &
+     prog_meta(iVar)%varName=='mLayerVolFracWat'          .or. &
+     prog_meta(iVar)%varName=='mLayerHeight'                   ) cycle
 
   ! get variable id
   err = nf90_inq_varid(ncID,trim(prog_meta(iVar)%varName),ncVarID); call netcdf_err(err,message)
+  if(err/=0)then
+   message=trim(message)//': problem with getting variable id, var='//trim(prog_meta(iVar)%varName)
+   return
+  endif
 
   ! get variable dimension IDs
   select case (prog_meta(iVar)%varType)
    case (iLookVarType%scalarv); err = nf90_inq_dimid(ncID,trim(scalDimName)   ,dimID); call netcdf_err(err,message)
-   case (iLookVarType%wlength); err = nf90_inq_dimid(ncID,trim(specDimName)   ,dimID); call netcdf_err(err,message)
    case (iLookVarType%midSoil); err = nf90_inq_dimid(ncID,trim(midSoilDimName),dimID); call netcdf_err(err,message)
    case (iLookVarType%midToto); err = nf90_inq_dimid(ncID,trim(midTotoDimName),dimID); call netcdf_err(err,message)
    case (iLookVarType%ifcToto); err = nf90_inq_dimid(ncID,trim(ifcTotoDimName),dimID); call netcdf_err(err,message)
-   case (iLookVarType%midSnow); err = nf90_inq_dimid(ncID,trim(midSnowDimName),dimID); call netcdf_err(err,message)
```

Comment:
State variables for snow are also state variables for soil and have the dimension `midToto`, This should work. Note that we return an error if `midSnow` is desired and not found.